config/config_mgmt.py

```diff
@@ -430,65 +478,84 @@ def mergeItems(it1, it2):
+                    if self.searchKeysInConfig(In[key], Out[key], skeys) == False:
+                        del Out[key]
+                    else:
+                        found = True
 
-        """
-        create Default Config using DFS for all ports
-        """
-        def createDefConfig(In, Out, ports):
-
-            found = False
-            if isinstance(In, dict):
-                for key in In.keys():
-                    #print("key:" + key)
-                    for port in ports:
-                        # pattern is very specific to current primary keys in
-                        # config DB, may need to be updated later.
-                        pattern = '^' + port + '\|' + '|' + port + '$' + \
-                            '|' + '^' + port + '$'
-                        #print(pattern)
-                        reg = re.compile(pattern)
-                        #print(reg)
-                        if reg.search(key):
-                            # In primary key, only 1 match can be found, so return
-                            # print("Added key:" + key)
-                            Out[key] = In[key]
-                            found = True
-                            break
-                    # Put the key in Out by default, if not added already.
-                    # Remove later, if subelements does not contain any port.
-                    if Out.get(key) is None:
-                        Out[key] = type(In[key])()
-                        if createDefConfig(In[key], Out[key], ports) == False:
-                            del Out[key]
-                        else:
-                            found = True
+        elif isinstance(In, list):
+            for skey in skeys:
+                if skey in In:
+                    found = True
+                    Out.append(skey)
+                    #print("Added in list:" + port)
 
-            elif isinstance(In, list):
-                for port in ports:
-                    if port in In:
-                        found = True
-                        Out.append(port)
-                        #print("Added in list:" + port)
+        else:
+            # nothing for other keys
+            pass
 
-            else:
-                # nothing for other keys
-                pass
+        return found
+
+    """
+    This function returns the relavant keys in Input Config.
+    For Example: All Ports related Config in Config DB.
+    """
+    def configWithKeys(self, configIn=dict(), keys=list()):
+
+        configOut = dict()
+        try:
+            if len(configIn) and len(keys):
+                self.searchKeysInConfig(configIn, configOut, skeys=keys)
+        except Exception as e:
+            print("configWithKeys Failed, Error: {}".format(str(e)))
+            raise e
 
-            return found
+        return configOut
+
+    """
+    Create a defConfig for given Ports from Default Config File.
+    """
+    def getDefaultConfig(self, ports=list()):
```

Comment:
This should have been more generic and should be used to get default config for any object. But we can think of it in later versions

Reply:
Under ConfigMgmtDPB class we do not want any other default config then related to the ports, so better to keep as ports as of now.